pages/settings/settings.js

```diff
@@ -500,3 +500,88 @@ proxyTypeInput.addEventListener('change', e => {
 })
 
 proxyInputs.forEach(item => item.addEventListener('change', e => setProxy(e.target.name, e.target.value)))
+
+settings.get('customBangs', (value) => {
+    const bangslist = document.getElementById('custom-bangs');
+    
+    if(value) { value.forEach(function(bang) {
+        bangslist.appendChild(createBang(bang.phrase, bang.snippet, bang.redirect))
+        bangslist.appendChild(document.createElement('br'));
+    })
+}
+})
+
+
+document.getElementById('add-custom-bang').onclick = function() {
+    const bangslist = document.getElementById('custom-bangs');
+    bangslist.appendChild(createBang());
+    bangslist.appendChild(document.createElement('br'));
+} 
+function createBang (bang, snippet, redirect) {
+    var li = document.createElement('li')
+    var bangInput = document.createElement('input')
+    var snippetInput = document.createElement('input')
+    var redirectInput = document.createElement('input');
+    var xButton = document.createElement('input');
+    var current = { phrase: bang ?? '', snippet: snippet ?? '', redirect: redirect ?? '' };
+    bangInput.type = 'text';
+    snippetInput.type = 'text';
+    redirectInput.type = 'text';
+    xButton.type = 'button';
+    bangInput.value = bang ?? '';
+    snippetInput.value = snippet ?? '';
+    redirectInput.value = redirect ?? '';
+    xButton.value = 'X';
```

Comment:
This should be an actual icon (like we discussed in Discord)

Reply:
Yup